File: PN30.cpp


#include<iostream>
using namespace std;

class Number
{
    private:
        int iNo;       

    public:
        int iCnt = 0;
		
        void Accept()       
        {
            cout<<"Enter the value : "<<endl;
            cin>>this->iNo;
        }

        int SumDigits()
		{
			int iDigit = 0;
			int iSum = 0;
			
			if(iNo > 0)
			{
				iNo = -iNo;
			}
			while(iNo > 0)
			{
				iDigit = iNo % 10;
				iSum = iSum + iDigit; 
				iNo = iNo /10;
			}
			return iSum;
			
		
		}
		
		
		

};

int main()
{
    Number nobj;
	int iRet = 0;
	


    nobj.Accept();

	iRet = nobj.SumDigits();
	cout<<iRet<<endl;


    return 0;
}

File: PN30.java
import java.lang.*;
import java.util.*;

class Number
{
    private int iNo;

    public void Accept()
    {
        Scanner sobj = new Scanner(System.in);
        System.out.println("Enter number : ");
        this.iNo = sobj.nextInt();
    }

    public void Display()
    {
        System.out.println("Value is : "+this.iNo);
    }

    public boolean Check()
    {
        if((iNo % 2) == 0)
		{
			return true;
		}
		else
		{
		    return false;
		}
			
			
           
    }
}

class PN30
{
    public static void main(String b[])
    {
            Number nobj = new Number();
            boolean iRet;

            nobj.Accept();
            nobj.Display();

            iRet = nobj.Check();
			
			if(iRet == true)
			{
				System.out.println("The number is even");
			}
			else 
			{
				System.out.println("The number is not even");
			}
            
    }
}
			